File: backend/internal/entity/onboarding.go
package entity

import "time"

type OnboardingRequest struct {
	GenreIDs  []int64  `json:"genre_ids"`
	ArtistIDs []int64  `json:"artist_ids"`
	TrackIDs  []int64  `json:"track_ids"`
	Contexts  []string `json:"contexts"`
}

type UserProfile struct {
	UserID            int64     `json:"user_id"`
	FavoriteGenreIDs  []int64   `json:"favorite_genre_ids"`
	FavoriteArtistIDs []int64   `json:"favorite_artist_ids"`
	StarterTrackIDs   []int64   `json:"starter_track_ids"`
	Contexts          []string  `json:"contexts"`
	UpdatedAt         time.Time `json:"updated_at"`
}

File: frontend/src/types/onboarding.ts
export type OnboardingRequest = {
  genre_ids: number[]
  artist_ids: number[]
  track_ids: number[]
  contexts: string[]
}

export type UserProfile = {
  user_id: number
  favorite_genre_ids: number[]
  favorite_artist_ids: number[]
  starter_track_ids: number[]
  contexts: string[]
  updated_at: string
}
